From 16af1bdbf5fc7293d4e0d75946aadf6b916e67a5 Mon Sep 17 00:00:00 2001
From: Trent Mick <trentm@gmail.com>
Date: Wed, 7 Dec 2016 22:54:28 +0000
Subject: [PATCH] joyent/node-lockfd#7 node v4 support (v1.x) Reviewed by: Cody
 Peter Mello <cody.mello@joyent.com> Approved by: Cody Peter Mello
 <cody.mello@joyent.com>

---
 CHANGES.md   |  5 +++++
 Makefile     | 10 +++++++---
 package.json |  4 ++--
 3 files changed, 14 insertions(+), 5 deletions(-)

diff --git a/CHANGES.md b/CHANGES.md
index 951655e..e0200e6 100644
--- a/CHANGES.md
+++ b/CHANGES.md
@@ -2,6 +2,11 @@
 
 ## not yet released
 
+(nothing yet)
+
+## 1.2.0
+
+- joyent/node-lockfd#7: Node v4 support
 - joyent/node-lockfd#6: *Removes* the flock(3C) support added in
   joyent/node-lockfd#5. See the master branch and 2.x releases for
   that functionality.
diff --git a/Makefile b/Makefile
index eea59d1..6ffedc8 100644
--- a/Makefile
+++ b/Makefile
@@ -30,13 +30,17 @@ LOCKFD_BINDING	:= ./lib/lockfd_binding.node
 #
 
 .PHONY: all
-all: $(LOCKFD_BINDING) ./node_modules
+all: $(LOCKFD_BINDING) | ./node_modules
 	$(NPM) install
 
-$(LOCKFD_BINDING):
+./node_modules:
+	$(NPM) install
+
+$(LOCKFD_BINDING): | ./node_modules
 	cd src && make
 
-CLEAN_FILES += $(LOCKFD_BINDING)
+clean:
+	rm -rf $(LOCKFD_BINDING) src/*.o node_modules
 
 # Ensure CHANGES.md and package.json have the same version.
 .PHONY: versioncheck
diff --git a/package.json b/package.json
index 8f65cef..0c8a796 100644
--- a/package.json
+++ b/package.json
@@ -1,12 +1,12 @@
 {
   "name": "lockfd",
   "description": "thin wrapper around fcntl(F_SETLK)",
-  "version": "1.0.1",
+  "version": "1.2.0",
   "author": "Joshua M. Clulow <jmc@joyent.com>",
   "repository": "git://github.com/joyent/node-lockfd.git",
   "main": "./lib/index.js",
   "dependencies": {
-    "v8plus": "~0.3.1"
+    "v8plus": "~1.0.2"
   },
   "scripts": {
     "postinstall": "cd src && gmake $(eval echo ${MAKE_OVERRIDES})",
-- 
2.21.0

